vulture/core.py

```diff
@@ -206,6 +236,24 @@ def exclude(name):
                     self.scan(module_string, filename=path)
 
     def report(self):
+        if self.weigh:
+            return self.report_weight_sorted()
+        else:
+            return self.report_alpha_sorted()
+
+    def report_weight_sorted(self):
+        unused_item_found = False
+        for item in sorted(
+                self.unused_funcs + self.unused_imports + self.unused_props +
+                self.unused_classes + self.unused_vars + self.unused_attrs,
+                key=lambda item: item.weight):
+            print("%s:%d: Unused %s '%s' weight:%d" % (
+                _format_path(item.filename), item.lineno,
+                item.typ, item, item.weight))
+            unused_item_found = True
+        return unused_item_found
```

Comment:
To reduce code duplication, I think we should do the reporting in one function and just add or omit the weight and adapt the sorting key according to `self.weigh`.